lib/embulk/guess/csv.rb

```diff
@@ -104,6 +104,11 @@ def guess_lines(config, sample_lines)
 
         sample_records = split_lines(parser_guessed, true, sample_lines, delim, {})
 
+        # It should fail if CSV parser cannot parse sample_lines.
+        if sample_records.nil? || sample_records.empty?
```

Comment:
Embulk's guess behavior is different by whether the input plugin can create a Buffer object or not.

1. Even though Embulk tries to guess for an empty file, the input plugin could not create any Buffer objects. It throws NoSampleException. If several files exist, Embulk tries to read them until it creates a Buffer object.
2. On the other hand, If Embulk tries to guess for a header-only file, the input plugin could create a Buffer object includes the header line. Therefore it could suggest something.

This PR fixes a part of 2nd one. The input plugin could a Buffer object but, CsvParserPlugin could not parse the lines in the Buffer object. In this case, v0.8.18 throws `NoMethodError` as mentioned in the PR's description. This fix enables avoiding the unexpected error and returning empty as the guess result.